### PR TITLE
Added support for URLs for `Video`, `Audio`, and `Image`

### PR DESCRIPTION
**Pull request by abidlabs**
Previously, you could only pass in local filepaths for the `Video` and `Audio` components. This PR adds support for URLs as well. 

Furthermore, while `Image` supported URLs for default values, examples with URLs didn't work. So this PR also fixes that. 

Fixes: #2254

### CODE REVIEW

**Comment by aliabd** [comment]
LGTM! Tested locally and works great :)

**Review by aliabd — APPROVED** [review]

**Comment by abidlabs** [comment]
Thanks for the review @aliabd! Just updated one of the example demos (reverse audio) to include an example URL. 

Once the CI tests pass, I'll merge it in

File: demo/reverse_audio/run.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ def reverse_audio(audio):
                     inputs="microphone", 
                     outputs="audio", 
                     examples=[
-                    os.path.join(os.path.dirname(__file__), "audio/cantina.wav"),
```
Comment (freddyaboulton) [inline demo/reverse_audio/run.py:17]: @abidlabs This doesn't work on spaces

**Comment by abidlabs** [comment]
@freddyaboulton it's working in the demos for this PR? https://huggingface.co/spaces/gradio-pr-deploys/pr-2256-all-demos

**Comment by abidlabs** [comment]
Have you pulled in all of the changes from `main` into that branch?

**Comment by freddyaboulton** [comment]
@abidlabs Going to factory reboot the space to see if that fixes it. I think that branch had to have the latest changes from main right? Otherwise the reverse_audio wouldn't have the file URL introduced in this pr.

**Comment by abidlabs** [comment]
Oh it looks like the file is no longer accessible! They probably have a timeout

**Comment by freddyaboulton** [comment]
@abidlabs Sounds good and thank you! Updated in this PR https://github.com/gradio-app/gradio/pull/2250

**Pull request merged**